src/ray/raylet/node_manager.cc

```diff
@@ -1782,7 +1782,13 @@ bool NodeManager::AssignTask(const Task &task) {
   std::shared_ptr<Worker> worker = worker_pool_.PopWorker(spec);
   if (worker == nullptr) {
     // There are no workers that can execute this task.
-    if (!spec.IsActorTask()) {
+    const bool need_start_new =
+        (!spec.WorkerStartingPrefix().empty() || !spec.WorkerStartingSuffix().empty());
+    if (spec.IsActorCreationTask() && need_start_new) {
+      if (!worker_pool_.HasWorkerStartingForTask(spec.GetLanguage(), spec.TaskId())) {
```

Comment:
Nit: `HasWorkerStarted`

Reply:
Nice catch!

I'd like to rename it to `IsWorkerStartingForTask()` because it queries whether  we're starting a worker for a specific actor creation task.